crates/common/src/raindex_client/add_orders.rs

```diff
@@ -59,18 +83,84 @@ impl RaindexClient {
     }
 }
 
+#[wasm_export]
+impl RaindexOrder {
+    /// Generates ABI-encoded calldata for the `addOrder2()` function on the orderbook contract
+    ///
+    /// Parses the dotrain configuration, extracts the specified deployment, and creates the
+    /// transaction calldata needed to add an order to the orderbook. The calldata includes
+    /// the order structure, evaluable bytecode, and all necessary parameters.
+    ///
+    /// ## Examples
+    ///
+    /// ```javascript
+    /// // Generate calldata for adding an order
+    /// const result = await order.getAddOrderCalldata(dotrainText, "flare");
+    /// if (result.error) {
+    ///   console.error('Failed:', result.error.readableMsg);
+    /// } else {
+    ///   const calldata = result.value;
+    ///   // Do something with the calldata
+    /// }
+    /// ```
+    #[wasm_export(
+        js_name = "getAddOrderCalldata",
+        return_description = "ABI-encoded calldata ready for blockchain submission",
+        unchecked_return_type = "Hex"
+    )]
+    pub async fn get_add_order_calldata(
```

Comment:
@findolor i don't really understand why this is an impl on RaindexOrder?

Reply:
yes this place is not correct for this function, will move the function to be on raindex client